omod/src/main/web/dashboardwidgets/programstatus/programstatus.controller.js

```diff
@@ -744,4 +747,58 @@ export default class ProgramStatusController {
         }
         return utcDate;
     }
+
+    patientProgramComparator(patientProgram1, patientProgram2) {
+        const dateEnrolled1 = new Date(patientProgram1.value.dateEnrolled);
+        const dateEnrolled2 = new Date(patientProgram2.value.dateEnrolled);
+
+        // first check enrollment dates
+        if (dateEnrolled1 > dateEnrolled2) {
+          return -1
+        }
+        else if (dateEnrolled1 < dateEnrolled2) {
+          return 1
+        }
+
+        // if enrollment dates are equal, check completion dates
+        const dateCompleted1 = patientProgram1.value.dateCompleted ? new Date(patientProgram1.value.dateCompleted) : null;
+        const dateCompleted2 = patientProgram2.value.dateCompleted ? new Date(patientProgram2.value.dateCompleted) : null;
+
+        // "active" (no completion date) ranks higher
+        // assumption: never should be two active programs (ie both date completed should not be null)
+        if (dateCompleted1 === null) {
+          return -1;
+        }
+        else if (dateCompleted2 === null) {
+          return 1;
+        }
+        // having two programs with the same start date but different end dates should be illegal, but just in case
+        else if (dateCompleted1 > dateCompleted2) {
+          return -1;
+        }
+        else if (dateCompleted1 < dateCompleted2) {
+          return 1;
+        }
+
+        // if still equal, fall back to date created
+        // if enrollment dates are equal, check completion dates
+        const dateCreated1 = patientProgram1.value.dateCreated;
+        const dateCreated2 = patientProgram2.value.dateCreated;
+
+        if (dateCreated1 > dateCreated2) {
+          return -1;
+        }
+        else if (dateCreated1 < dateCreated2) {
+          return 1;
+        }
+
+        // and finally, just use uuid, because we need to make sure the sort order is deterministic
+        if (patientProgram1.value.uuid < patientProgram2.value.uuid) {
```

Comment:
Shouldn't this be consistent with what is in the java code?  Either both use "id" or both use "uuid"?